packages/ra-core/src/controller/list/useListParams.ts

```diff
@@ -157,6 +157,11 @@ export const useListParams = ({
                 tempParams.current = queryReducer(query, action);
                 // schedule side effects for next tick
                 setTimeout(() => {
+                    if (!tempParams.current) {
+                        throw new Error(
+                            'Race condition in changeParams detected'
```

Comment:
Users might need more information to solve this

Reply:
I agree, but this error should never occur (check the source code). To be honest, I added it just to please TypeScript.

Reply:
Now that you mention it, throwing an error may break the app in legitimate cases (e.g. two successive changeParams calls). I've converted it to an early return.